app/mailers/discovered_mailer.rb

```diff
@@ -1,7 +1,14 @@
 class DiscoveredMailer < ::ApplicationMailer
   helper :discovered_hosts
   def discovered_summary(options = {})
-    raise ::Foreman::Exception.new(N_("Must specify a user with email enabled")) unless (user = User.find(options[:user])) && user.mail_enabled?
+    user = if options[:user].kind_of? User
+             options[:user]
+           elsif options[:user].kind_of? Integer
+             User.find(options[:user])
+           else
+             raise ::Foreman::Exception.new(_('Invalid provided user type of %s ') % options[:user].class.to_s)
```

Comment:
I'd probably include the fact the exception was raised during mail generation: "Unable to generate discovery summary: invalid..."

Reply:
But the email is for a list of facts, and it's before you are looking for them, so it does not make sense to write a list of them imo.